# S30 - Evidencia de aprendizaje
Comprender de forma práctica y aplicada el manejo de datos utilizando Pandas.

##  1. Crea un DataFrame frutas que luzca así:

![Reto Uno]()

In [1]:
import pandas as pd

datos = {
    'Granadilla': pd.Series([20]),
    'Tomates': pd.Series([50])
    }
frutas = pd.DataFrame(datos)
print(frutas)


   Granadilla  Tomates
0          20       50


Creamos un diccionario con dos claves: 'Granadilla' y 'Tomates', cada una con una serie de Pandas que tiene un solo valor. Luego, convertimos el diccionario en un DataFrame con pd.DataFrame(datos) y lo mostramos con print(frutas).

##  2. Crea un DataFrame ventas_frutas que coincida con el diagrama:

![reto dos]()

In [4]:
import pandas as pd

datos = {
    'Granadilla': pd.Series([20, 49], index=["ventas 2021","ventas 2022"]),
    'Tomates': pd.Series([50, 100], index=["ventas 2021","ventas 2022"])
    }
ventas_frutas = pd.DataFrame(datos)
print(ventas_frutas)

             Granadilla  Tomates
ventas 2021          20       50
ventas 2022          49      100


Creamos un diccionario con dos claves: 'Granadilla' y 'Tomates', donde cada una tiene una serie de Pandas con datos de ventas para los años 2021 y 2022.

Usamos el parámetro index para asignar etiquetas a las filas. Luego, convertimos el diccionario en un DataFrame con pd.DataFrame(datos) y lo mostramos con print(ventas_frutas).

## 3. Crea una variable utensilios con una Serie que tenga el siguiente aspecto:

![reto tres]()

In [6]:
import pandas as pd

utensilios = pd.Series(["3 unidades",
                        "2 unidades",
                        "4 unidades",
                        "5 unidades"], index=["Cuchara",
                                              "Tenedor",
                                              "Cuchillo",
                                              "Plato"], name="Cocina")
print(utensilios)

Cuchara     3 unidades
Tenedor     2 unidades
Cuchillo    4 unidades
Plato       5 unidades
Name: Cocina, dtype: object


Se crea una serie con pd.Series(), donde se ponen los números de cada utensilio. Se le dan nombres con index y se le pone un nombre a la serie con name="Cocina". Al final, se imprime con print().

## 4. Descarga el dataset 'wine review' desde Kaggle y cárgalo en un DataFrame llamado review, tal y como se muestra en la figura.

![reto cuatro]()

In [7]:
import kaggle

download_path = '.kaggle/wine_mag'

kaggle.api.dataset_download_files('christopheiv/winemagdata130k', path=download_path, unzip=True)
wine_review_api = pd.read_csv(download_path+"/winemag-data-130k-v2.csv", index_col=0)

wine_review_api

Dataset URL: https://www.kaggle.com/datasets/christopheiv/winemagdata130k


country                                        description  \
0          Italy  Aromas include tropical fruit, broom, brimston...   
1       Portugal  This is ripe and fruity, a wine that is smooth...   
2             US  Tart and snappy, the flavors of lime flesh and...   
3             US  Pineapple rind, lemon pith and orange blossom ...   
4             US  Much like the regular bottling from 2012, this...   
...          ...                                                ...   
129966   Germany  Notes of honeysuckle and cantaloupe sweeten th...   
129967        US  Citation is given as much as a decade of bottl...   
129968    France  Well-drained gravel soil gives this wine its c...   
129969    France  A dry style of Pinot Gris, this is crisp with ...   
129970    France  Big, rich and off-dry, this is powered by inte...   

                                   designation  points  price  \
0                                 Vulkà Bianco      87    NaN   
1                                     Avidagos      87   15.0   
2                                          NaN      87   14.0   
3                         Reserve Late Harvest      87   13.0   
4           Vintner's Reserve Wild Child Block      87   65.0   
...                                        ...     ...    ...   
129966  Brauneberger Juffer-Sonnenuhr Spätlese      90   28.0   
129967                                     NaN      90   75.0   
129968                                   Kritt      90   30.0   
129969                                     NaN      90   32.0   
129970           Lieu-dit Harth Cuvée Caroline      90   21.0   

                 province             region_1           region_2  \
0       Sicily & Sardinia                 Etna                NaN   
1                   Douro                  NaN                NaN   
2                  Oregon    Willamette Valley  Willamette Valley   
3                Michigan  Lake Michigan Shore                NaN   
4                  Oregon    Willamette Valley  Willamette Valley   
...                   ...                  ...                ...   
129966              Mosel                  NaN                NaN   
129967             Oregon               Oregon       Oregon Other   
129968             Alsace               Alsace                NaN   
129969             Alsace               Alsace                NaN   
129970             Alsace               Alsace                NaN   

               taster_name taster_twitter_handle  \
0            Kerin O’Keefe          @kerinokeefe   
1               Roger Voss            @vossroger   
2             Paul Gregutt           @paulgwine    
3       Alexander Peartree                   NaN   
4             Paul Gregutt           @paulgwine    
...                    ...                   ...   
129966  Anna Lee C. Iijima                   NaN   
129967        Paul Gregutt           @paulgwine    
129968          Roger Voss            @vossroger   
129969          Roger Voss            @vossroger   
129970          Roger Voss            @vossroger   

                                                    title         variety  \
0                       Nicosia 2013 Vulkà Bianco  (Etna)     White Blend   
1           Quinta dos Avidagos 2011 Avidagos Red (Douro)  Portuguese Red   
2           Rainstorm 2013 Pinot Gris (Willamette Valley)      Pinot Gris   
3       St. Julian 2013 Reserve Late Harvest Riesling ...        Riesling   
4       Sweet Cheeks 2012 Vintner's Reserve Wild Child...      Pinot Noir   
...                                                   ...             ...   
129966  Dr. H. Thanisch (Erben Müller-Burggraef) 2013 ...        Riesling   
129967                  Citation 2004 Pinot Noir (Oregon)      Pinot Noir   
129968  Domaine Gresser 2013 Kritt Gewurztraminer (Als...  Gewürztraminer   
129969      Domaine Marcel Deiss 2012 Pinot Gris (Alsace)      Pinot Gris   
129970  Domaine Schoffit 2012 Lieu-dit Harth Cuvée Car...  Gewürztraminer 

Usamos kaggle.api.dataset_download_files para bajar los datos y después los leemos con pandas para verlos.

## 5. Visualiza las primeras filas del DataFrame 'wine review':

![reto cinco]()

In [10]:
wine_review_api.head()

country                                        description  \
0     Italy  Aromas include tropical fruit, broom, brimston...   
1  Portugal  This is ripe and fruity, a wine that is smooth...   
2        US  Tart and snappy, the flavors of lime flesh and...   
3        US  Pineapple rind, lemon pith and orange blossom ...   
4        US  Much like the regular bottling from 2012, this...   

                          designation  points  price           province  \
0                        Vulkà Bianco      87    NaN  Sicily & Sardinia   
1                            Avidagos      87   15.0              Douro   
2                                 NaN      87   14.0             Oregon   
3                Reserve Late Harvest      87   13.0           Michigan   
4  Vintner's Reserve Wild Child Block      87   65.0             Oregon   

              region_1           region_2         taster_name  \
0                 Etna                NaN       Kerin O’Keefe   
1                  NaN                NaN          Roger Voss   
2    Willamette Valley  Willamette Valley        Paul Gregutt   
3  Lake Michigan Shore                NaN  Alexander Peartree   
4    Willamette Valley  Willamette Valley        Paul Gregutt   

  taster_twitter_handle                                              title  \
0          @kerinokeefe                  Nicosia 2013 Vulkà Bianco  (Etna)   
1            @vossroger      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2           @paulgwine       Rainstorm 2013 Pinot Gris (Willamette Valley)   
3                   NaN  St. Julian 2013 Reserve Late Harvest Riesling ...   
4           @paulgwine   Sweet Cheeks 2012 Vintner's Reserve Wild Child...   

          variety               winery  
0     White Blend              Nicosia  
1  Portuguese Red  Quinta dos Avidagos  
2      Pinot Gris            Rainstorm  
3        Riesling           St. Julian  
4      Pinot Noir         Sweet Cheeks

Utilizamos el metodo head(), por defecto este metodo nos trae los primeros cinco(5) registros

## 6. Utiliza el método .info() para averiguar cuántas entradas hay.
## ¿Cuántas encontraste?



In [11]:
wine_review_api.info()

<class 'pandas.core.frame.DataFrame'>
Index: 129971 entries, 0 to 129970
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   country                129908 non-null  object 
 1   description            129971 non-null  object 
 2   designation            92506 non-null   object 
 3   points                 129971 non-null  int64  
 4   price                  120975 non-null  float64
 5   province               129908 non-null  object 
 6   region_1               108724 non-null  object 
 7   region_2               50511 non-null   object 
 8   taster_name            103727 non-null  object 
 9   taster_twitter_handle  98758 non-null   object 
 10  title                  129971 non-null  object 
 11  variety                129970 non-null  object 
 12  winery                 129971 non-null  object 
dtypes: float64(1), int64(1), object(11)
memory usage: 13.9+ MB


se encontraron 129971 entradas o de 0 a 129970 entradas


## 7. ¿Cuál es el precio promedio?

In [8]:
precio_promedio = wine_review_api['price'].mean() 
print(f"El precio promedio es: {precio_promedio}")

El precio promedio es: 35.363389129985535


con el metodo mean() encontramos el promedio en la culumna de ["price"]

## 8. ¿Cuál es el precio más alto pagado?

In [9]:
precio_maximo = wine_review_api['price'].max()
print(f"El precio más alto pagado es: {precio_maximo}")

El precio más alto pagado es: 3300.0


con el metodo max() encontramos el precio maximo dentro de ["price"]

## 9. Crea un DataFrame con todos los vinos de california.
## Salida esperada:

![reto nueve]()

In [10]:
vinos_california = wine_review_api[wine_review_api['province'] == 'California']

vinos_california

country                                        description  \
10          US  Soft, supple plum envelopes an oaky structure ...   
12          US  Slightly reduced, this wine offers a chalky, t...   
14          US  Building on 150 years and six generations of w...   
23          US  This wine from the Geneseo district offers aro...   
25          US  Oak and earth intermingle around robust aromas...   
...        ...                                                ...   
129942      US  This is classic in herbaceous aromas and flavo...   
129945      US  Hailing from one of the more popular vineyards...   
129949      US  There's no bones about the use of oak in this ...   
129950      US  This opens with herbaceous dollops of thyme an...   
129952      US  This Zinfandel from the eastern section of Nap...   

                             designation  points  price    province  \
10                        Mountain Cuvée      87   19.0  California   
12                                   NaN      87   34.0  California   
14                                   NaN      87   12.0  California   
23                   Signature Selection      87   22.0  California   
25                   King Ridge Vineyard      87   69.0  California   
...                                  ...     ...    ...         ...   
129942                               NaN      90   35.0  California   
129945  Jurassic Park Vineyard Old Vines      90   20.0  California   
129949                  Barrel Fermented      90   35.0  California   
129950                     Blocks 7 & 22      90   35.0  California   
129952                               NaN      90   22.0  California   

                 region_1       region_2     taster_name  \
10            Napa Valley           Napa  Virginie Boone   
12       Alexander Valley         Sonoma  Virginie Boone   
14          Central Coast  Central Coast   Matt Kettmann   
23            Paso Robles  Central Coast   Matt Kettmann   
25           Sonoma Coast         Sonoma  Virginie Boone   
...                   ...            ...             ...   
129942      Sonoma County         Sonoma  Virginie Boone   
129945  Santa Ynez Valley  Central Coast   Matt Kettmann   
129949        Napa Valley           Napa  Virginie Boone   
129950        Napa Valley           Napa  Virginie Boone   
129952      Chiles Valley           Napa  Virginie Boone   

       taster_twitter_handle  \
10                   @vboone   
12                   @vboone   
14             @mattkettmann   
23             @mattkettmann   
25                   @vboone   
...                      ...   
129942               @vboone   
129945         @mattkettmann   
129949               @vboone   
129950               @vboone   
129952               @vboone   

                                                    title             variety  \
10      Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
12      Louis M. Martini 2012 Cabernet Sauvignon (Alex...  Cabernet Sauvignon   
14               Mirassou 2012 Chardonnay (Central Coast)          Chardonnay   
23      Bianchi 2011 Signature Selection Merlot (Paso ...              Merlot   
25      Castello di Amorosa 2011 King Ridge Vineyard P...          Pinot Noir   
...                                                   ...                 ...   
129942   Arrowood 2010 Cabernet Sauvignon (Sonoma County)  Cabernet Sauvignon   
129945  Birichino 2013 Jurassic Park Vineyard Old Vine...        Chenin Blanc   
129949  Flora Springs 2013 Barrel Fermented Chardonnay...          Chardonnay   
129950  Hendry 2012 Blocks 7 & 22 Zinfandel (Napa Valley)           Zinfandel   
129952             Houdini 2011 Zinfandel (Chiles Valley)           Zinfandel   

                     winery  
10       Kirkland Signature  
12         Louis M. Martini  
14                 Mirassou  
23                  Bianchi  
25      Castello di Amorosa  
...                     ...  
129942             Arrowood  
129945

Con wine_review_api['province'] == 'California' seleccionamos solo las filas donde la provincia es "California". Luego guardamos el resultado en vinos_california y lo mostramos.

## 10. Utiliza idxmax() para encontrar el índice del vino con el precio más alto y luego utiliza loc para obtener toda la información de ese vino específico.
## Salida esperada:

![reto diez]()

In [29]:
indice_vino_caro = wine_review_api['price'].idxmax() 

indice_vino_caro_info = wine_review_api.loc[indice_vino_caro] 

indice_vino_caro_info

country                                                             France
description              This ripe wine shows plenty of blackberry frui...
designation                                                            NaN
points                                                                  88
price                                                               3300.0
province                                                          Bordeaux
region_1                                                             Médoc
region_2                                                               NaN
taster_name                                                     Roger Voss
taster_twitter_handle                                           @vossroger
title                                 Château les Ormes Sorbet 2013  Médoc
variety                                           Bordeaux-style Red Blend
winery                                            Château les Ormes Sorbet
Name: 80290, dtype: objec

idxmax() nos da el índice donde el precio es máximo, y loc[] nos permite obtener toda la fila con la información del vino más caro.

## 11.  ¿Cuáles son los tipos de uva más comunes en California?
## Salida esperada:

![reto 11]()

In [ ]:
vinos_california = wine_review_api[wine_review_api['province'] == 'California'] 

uvas_comunes = vinos_california['variety'].value_counts() 

uvas_comunes

variety
Pinot Noir                        6896
Cabernet Sauvignon                5693
Chardonnay                        5183
Zinfandel                         2639
Syrah                             1870
                                  ... 
Tempranillo-Cabernet Sauvignon       1
Syrah-Grenache-Viognier              1
Negroamaro                           1
Moscato Giallo                       1
Alvarinho                            1
Name: count, Length: 194, dtype: int64

value_counts() nos muestra la cantidad de veces que aparece cada tipo de uva en los vinos de California.

## 12. ¿Cuáles son los 10 tipos de uva más comunes en California?
## Salida esperada:

![reto 12]()

In [34]:
uvas_comunes.head(10)

variety
Pinot Noir                  6896
Cabernet Sauvignon          5693
Chardonnay                  5183
Zinfandel                   2639
Syrah                       1870
Sauvignon Blanc             1807
Red Blend                   1804
Merlot                      1391
Bordeaux-style Red Blend     876
Petite Sirah                 694
Name: count, dtype: int64

Con head(10) traemos los 10 primeros registros